### PR TITLE
Trying to upgrade torchgeo

### PR DESCRIPTION
**Pull request by Joao-L-S-Almeida**
torchgeo >= 0.6.0

### CODE REVIEW

**Comment by CarlosGomes98** [comment]
The changes on requirements.txt are fine. But pyproject should try to be as lenient as possible so that `terratorch` plays nicely with as many libraries as possible. If a library out there requires torchgeo-0.5.2 for some reason, we would still like to be compatible with them. Basically it should have the lowest version we tolerate in order for all the functionalities to work. We should only change this part if we introduce a feature that really requires the newest torchgeo.

Like this, when someone installs terratorch, `pip` will install the newest possible `torchgeo` that is still compatible with all the other requirements the user has.

File: pyproject.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ classifiers = [
 dependencies = [
   "torch>=2.1.0",
   "torchvision>=0.16.0",
-  "torchgeo>=0.5.1",
+  "torchgeo>=0.6.0",
```
Comment (CarlosGomes98) [inline pyproject.toml:34]:
```suggestion
  "torchgeo>=0.5.1",
```

**Review by CarlosGomes98 — COMMENTED** [review]
see above

**Comment by Joao-L-S-Almeida** [comment]
@CarlosGomes98 Done.

**Review by CarlosGomes98 — APPROVED** [review]

**Pull request merged**